### PR TITLE
Remove import macro and improve build macro

### LINKED ISSUE #268: Change `build!` macro to be inline.
Currently in build scripts the `winrt::build!` macro generates a build function which the user then calls. This was because until the latest version of Rust, proc-macros were not allowed in expression position. With the latest release of Rust, the following would be possible in a build.rs:

```rust
fn main() {
    winrt::build!()
}
```

The only downside is we can only support the latest version of Rust. Given that we've not really made strong minimum supported Rust version (MSRV) claims, I think this is a fine tradeoff to make.

### LINKED ISSUE #304: Simplify import/build dependency declarations
The `os` dependency is provided by the `build`/`import` macros as follows:

```rust
winrt::import!(
    dependencies
        os
    types
        ...
);
```

This dependency is a shorthand for the winmd files in the `Windows\System32\WinMetadata` folder and is convenient for bootstrapping since it provides a pretty complete set of metadata for the Windows API. The trouble is that it is specific to the version of the build machine and thus not a stable OS version that's independent of where the code happens to be compiled. 

Now that Rust/WinRT supports nuget dependencies and the metadata is available via the `Microsoft.Windows.SDK.Contracts` nuget package, we should consider dropping the `os` dependency and simplify the dependency logic accordingly.

We should also consider removing the redundancy of dependency declarations. 

Rust dependencies are declared in the `Cargo.toml` file and Rust paths in `.rs` files are then used to make names from those dependencies available. That's the Rust way of doing things.

Right now, WinRT dependencies are listed in the `Cargo.toml` file as follows:

```
[package.metadata.winrt.dependencies]
"Microsoft.Windows.SDK.Contracts" = "10.0.19041.1"
"KennyKerr.Windows.TestWinRT" = "1.0.21"
```

And then repeated in the `build`/`import` macros:

```rust
winrt::build!(
    dependencies
        nuget: Microsoft.Windows.SDK.Contracts
        nuget: KennyKerr.Windows.TestWinRT
    types
       windows::foundation::*
       test_component::*
);
```

We should just skip this second step and infer dependencies from the original declarations in the `Cargo.toml` file.

### PR DESCRIPTION
**Pull request by rylev**
Fixes #304 
Fixes #268 

Some highlights:
* Create a new crate for resolving WinRT dependencies. It's unfortunate we need another crate, but we would otherwise need to duplicate quite a lot of code between `cargo winrt` and the `winrt_macros` crate. 
* Creates a integration_tests crate which is used for integration tests against the Windows SDK and test component
* Removes the `import!` macro and the ability to depend on the OS for metadata.

### CODE REVIEW

**Comment by kennykerr** [comment]
Looks promising!

**Comment by rylev** [comment]
@kennykerr this is ready for a more in-depth review!

**Comment by kennykerr** [comment]
Will do.

File: README.md [hunk 1]
```diff
@@ -19,22 +19,43 @@ winrt = "0.7.2"
 
 This will allow Cargo to download, build, and cache the Rust/WinRT support as a package directly from crates.io.
 
+
+Next, specifiy your WinRT dependencies in your Cargo.toml file:
```
Comment (kennykerr) [inline README.md:23]: typo

File: README.md [hunk 2]
```diff
@@ -19,22 +19,43 @@ winrt = "0.7.2"
...
+}
+```
+
+Then use those types inside your crate.
+
+```rust
+use winrt::*;
+
+mod bindings {
+    include_bindings!();
```
Comment (kennykerr) [inline README.md:52]: Nice.

File: crates/winrt-deps/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+mod error;
+pub mod manifest;
+
+use std::path::PathBuf;
+
+#[derive(Debug)]
+pub enum DependencyDescriptor {
+    NugetOrg { name: String, version: String },
+    Url { name: String, url: String },
+    Local { name: String, path: PathBuf },
```
[the remark this answers is not in the document]
Comment (rylev) [reply crates/winrt-deps/src/lib.rs:11]: In the interest of not letting this PR get too big, do you mind if we make issues for these and tackle them in new PRs?

[the remark this answers is not in the document]
Comment (kennykerr) [reply crates/winrt-deps/src/lib.rs:11]: Sounds good

[the remark this answers is not in the document]
Comment (rylev) [reply crates/winrt-deps/src/lib.rs:11]: Add two issues: https://github.com/microsoft/winrt-rs/issues/319 https://github.com/microsoft/winrt-rs/issues/320

**Pull request merged**